src/DataModel/IApiDocumentation.ts

```diff
@@ -0,0 +1,9 @@
+import {IMetadataDescription} from "./IMetadataDescription";
+import {IData} from "./IData";
+
+export interface IApiDocumentation extends IMetadataDescription
+{
+    entryPoint: string;
+
+    getEntryPoint(): Promise<IData>;
```

Comment:
> I wouldn't like to see developers create instances

Funny you'd say that. Previously I didn't actually implement it that way ;). Now that I look again at your code, maybe I was misjudging it a little.

My actual point is that `ApiDocumentation` seems to work both as a Hydra client and the documentation. I guess I'd simply move all the metadata business to `this.client` and maybe that's it